lib/configurator.js

```diff
@@ -27,6 +28,19 @@ function validateOptions (options) {
   )
 }
 
+var defaultSSCCookieDomain = function () {
+  var domainChunks = document.location.hostname.split('.')
+  for (var i = domainChunks.length - 2; i >= 1; i--) {
+    var domain = domainChunks.slice(i).join('.')
+    var name = '_td_domain_' + domain // append domain name to avoid race condition
+    cookie.setItem(name, domain, 3600, '/', domain)
+    if (cookie.getItem(name) === domain) {
+      return domain
+    }
+  }
+  return document.location.hostname
+}
```

Comment:
Does this do the right thing when the TLD has a `.` in it? It looks like it depends on the behaviour of `cookie.setItem` and `cookie.getItem`.